xdsl/printer.py

```diff
@@ -534,10 +536,25 @@ def print_attribute(self, attribute: Attribute) -> None:
 
         if isinstance(attribute, FloatAttr):
             value = attribute.value
-            attr_type = cast(
-                FloatAttr[Float16Type | Float32Type | Float64Type], attribute
-            ).type
-            self.print_string(f"{value.data:.6e} : ")
+            attr_type = cast(AnyFloatAttr, attribute).type
+            if math.isnan(value.data) or math.isinf(value.data):
+                if isinstance(attr_type, Float16Type):
+                    self.print_string(
+                        f"{hex(struct.unpack('<H', struct.pack('<e', value.data))[0])} : "
+                    )
+                elif isinstance(attr_type, Float32Type):
+                    self.print_string(
+                        f"{hex(struct.unpack('<I', struct.pack('<f', value.data))[0])} : "
+                    )
+                elif isinstance(attr_type, Float64Type):
+                    self.print_string(
+                        f"{hex(struct.unpack('<Q', struct.pack('<d', value.data))[0])} : "
+                    )
+                else:
+                    # todo add support for further bitwidths
+                    self.print_string(f"{value.data:.6e} : ")
```

Comment:
I'm wondering if this should throw an error

Reply:
Yeah seems like the right thing to do! Also might want to refactor this to a match statement, and throw an error in the default branch

Reply:
Raising it as an error.